app/helpers/chat_channels_helper.rb

```diff
@@ -0,0 +1,48 @@
+# rubocop:disable Rails/HelperInstanceVariable
+module ChatChannelsHelper
+  def render_unopened_json_response
+    @chat_channels_memberships = if session_current_user_id
+                                   ChatChannelMembership.where(user_id: session_current_user_id)
+                                     .where(has_unopened_messages: true)
+                                     .where(show_global_badge_notification: true)
+                                     .where.not(status: %w[removed_from_channel left_channel])
+                                     .includes(%i[chat_channel user])
+                                     .order("chat_channel_memberships.updated_at" => :desc)
+                                 else
+                                   []
```

Comment:
Non-blocking: It would be slight more Rails-y to use `ChatChannelMembership.none` here, so `@chat_channels_memberships` is always an instance of `ChatChannelMembership::ActiveRecord_Relation`.